doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-ESP8266/index.md

```diff
@@ -5,4 +5,383 @@ permalink: /:path/
+                           ${ESP_RTOS_SDK}/include/lwip/ipv4
+                           ${ESP_RTOS_SDK}/include/lwip/ipv6
+                           ${ESP_RTOS_SDK}/include/espressif/
+                           )
+~~~
+
+We also need to use custom linked script.
+Below, two linker scripts, required for ESP8266 applications, are moved to binary directory so that linker can locate them.
+
+~~~CMake
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/ld/eagle.rom.addr.v6.ld ${CMAKE_BINARY_DIR})
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/ld/eagle.app.v6.ld ${CMAKE_BINARY_DIR})
+~~~
+
+Next, we should tell CMake what libraries we would like to link with.
+Here the required libraries from ESP8266 RTOS SDK, linker script and Kaa SDK are specified.
+
+~~~CMake
+exec_program(xtensa-lx106-elf-gcc .
+            ARGS -print-libgcc-file-name
+            OUTPUT_VARIABLE ESP8266_LIBGCC
+            )
+
+target_link_libraries(kaa_demo_s PUBLIC
+                      kaac
+                      ${ESP_RTOS_SDK}/lib/libfreertos.a
+                      ${ESP_RTOS_SDK}/lib/libhal.a
+                      ${ESP_RTOS_SDK}/lib/libpp.a
+                      ${ESP_RTOS_SDK}/lib/libphy.a
+                      ${ESP_RTOS_SDK}/lib/libnet80211.a
+                      ${ESP_RTOS_SDK}/lib/libwpa.a
+                      ${ESP_RTOS_SDK}/lib/liblwip.a
+                      ${ESP_RTOS_SDK}/lib/libmain.a
+                      ${ESP_RTOS_SDK}/lib/libssl.a
+                      ${ESP_RTOS_SDK}/lib/libhal.a
+                      ${ESP8266_LIBGCC}
+                      -T${CMAKE_CURRENT_SOURCE_DIR}/ld/eagle.app.v6.ld
+                      )
+~~~
+
+This is required due to ESP8266 specific requirements regarding linked executable.
+The blank.c file is a placeholder for CMake's `add_executable()`.
+All the code (Kaa SDK, ESP8266 SDK and demo) is compiled as static libraries
+and linked into that executable.
+
+~~~CMake
+file(WRITE ${CMAKE_BINARY_DIR}/blank.c "")
+add_executable(kaa_demo ${CMAKE_BINARY_DIR}/blank.c)
+~~~
+
+Finally, we link our application with our demo library.
+
+~~~
+target_link_libraries(kaa_demo kaa_demo_s);
+~~~
+
+## Building
+To actually invoke CMake, proceed as follows:
+
+        mkdir build
+        cd build
+        cmake .. \
+            -DCMAKE_TOOLCHAIN_FILE=../libs/kaa/toolchains/esp8266.cmake \
+            -DKAA_PLATFORM=esp8266 \
+            -DCMAKE_BUILD_TYPE=MinSizeRel \
+            -DKAA_WITH_EXTENSION_CONFIGURATION=OFF \
+            -DKAA_WITH_EXTENSION_EVENT=OFF \
+            -DKAA_WITH_EXTENSION_NOTIFICATION=OFF \
+            -DKAA_WITH_EXTENSION_LOGGING=OFF \
+            -DKAA_WITH_EXTENSION_USER=OFF \
+            -DKAA_WITH_EXTENSION_PROFILE=OFF \
+            -DKAA_MAX_LOG_LEVEL=3
+
+This will generate `Makefile` in `build` directory. To actually build application invoke `make`.
+
+As you see, we should pass some arguments to CMake in order to build application for ESP8266:
+
+ Name | Description
+------|-------------
```

Comment:
There is no need to describe these parameters here. It should be described on the top-level c-sdk description. Also I'd exclude cmake's variables description.